episodes/07-thresholding.md

```diff
@@ -627,21 +630,22 @@ def enhanced_root_mass(filename, sigma):
 
     # perform binary thresholding to mask the white label and circle
     binary_mask = blurred_image < 0.95
-    # use the mask to remove the circle and label from the blurred image
-    blurred_image[~binary_mask] = 0
-
-    # perform automatic thresholding to produce a binary image
-    t = ski.filters.threshold_otsu(blurred_image)
-    binary_mask = blurred_image > t
+    
+    # perform automatic thresholding using only the pixels with value True in the binary mask
+    t = ski.filters.threshold_otsu(blurred_image[binary_mask])
+    
+    # update binary mask to identify pixels which are both less than 0.95 and greater than t
+    binary_mask = np.logical_and(binary_mask, blurred_image > t)
```

Comment:
I do like this solution, however, I am concerned that the introduction of `np.logical_and` may increase cognitive load. An alternative might be 
```binary_mask = (t < blurred_image) & (blurred_image < 0.95)```
but this would also require understanding of Boolean logic, which is only skimmed over in earlier episodes. Any thoughts from @datacarpentry/image-processing-curriculum-maintainers or @datacarpentry/curriculum-advisors-image?

Reply:
Perhaps we can add a `spoiler` block with a brief explanation of AND in this context? I will propose something...

Reply:
See [my suggestion](https://github.com/datacarpentry/image-processing/pull/315/files#r1472764124) below

Reply:
I think it's not bad to introduce a bit of Boolean logic when using a binary mask... But, maybe, if we wanted to avoid it, we could use compose the two masks?
```py
mask_white = blurred_image < 0.95
mask_thresh = blurred_image > t
mask_total = mask_thresh[mask_white]
```
And use the 'total' mask for selection? So we're not introducing new syntax.

Reply:
I think combining masks in this way is something that learners will use a lot if they carry on doing image proccessing/analysis after the course so worth introducing. Personally I think @uschille solution is the easiest to understand of the three so Im going with that.